mitodl/ocw-hugo-themes
# Update webpack devserver; convert webpack conifgs to ts

#### Pre-Flight checklist


#### What are the relevant tickets?
On the road to #919 

#### What's this PR do?
This PR updates webpack-dev-server to version 4 and changes the configuration files from js to typescript, because then we get waaaay better autocomplete on webpack stuff.

#### How should this be manually tested?
Run `yarn start:course`, `yarn start:www`, yarn `build:webpack` locally. They should work.

#### Any background context you want to provide?
I think the test setup for #919 will want to run webpack programmatically (ie, not through CLI). The docs I found easily for that were for webpack-dev-server v4. Rather than finding the old docs, I figure let's update the dependency.
